src/main/java/tech/sourced/gemini/WeightedMinHash.java

```diff
@@ -77,7 +77,7 @@ public WeightedMinHash(int dim, int sampleSize, long seed) {
         }
     }
 
-    protected WeightedMinHash(int dim, int sampleSize, double[][] rs, double[][] lnCs, double[][] betas) {
+    public WeightedMinHash(int dim, int sampleSize, double[][] rs, double[][] lnCs, double[][] betas) {
```

Comment:
A nitpick, but in cases like this it's usually better to widen the access modifier to a default one, which is a _package-private_ and should work in this case, by just deleting `protected`.

Reply:
default one should be `WeightedMinHash(int dim)` or `WeightedMinHash(int dim, int sampleSize, long seed)`. I use this one temporary while we don't have our own hash.